python/src/aiconfig/default_parsers/openai.py

```diff
@@ -493,9 +496,21 @@ def add_prompt_as_message(
     output = aiconfig.get_latest_output(prompt)
     if output:
         if output.output_type == "execute_result":
-            output_message = output.data
-            if output_message["role"] == "assistant":
-                messages.append(output_message)
+            if output.metadata.get("role", "") == "assistant":
+                output_data = output.data
+                if isinstance(output_data, str):
+                    output_message = {
+                        "content": output_data,
+                        "role": "assistant",
+                    }
+                    function_call = output.metadata.get("function_call", None)
```

Comment:
I'm worried this logic is more brittle now than the previous. If we store raw response in metadata, we could probably use that instead? But ok for now